pkg/kubectl/run.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+)
+
+type BasicReplicationController struct{}
+
+func (BasicReplicationController) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"name", true},
+		{"replicas", true},
+		{"image", true},
+		{"command", false},
+	}
+}
+
+func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+	labelString, found := params["labels"]
+	var labels map[string]string
+	if found && len(labelString) > 0 {
+		labels = ParseLabels(labelString)
+	} else {
+		labels = map[string]string{
+			"name": params["name"],
```

Comment:
I wrote the wrong thing at first, I updated it to: labels.Set{"name":
name}.AsSelector()

S

On Tue, Jan 13, 2015 at 1:45 PM, Brendan Burns <notifications@github.com>
wrote:

> In pkg/kubectl/run.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3424#discussion-diff-22897598>
> :
>
> > +		{"labels", false},
> > +		{"name", true},
> > +		{"replicas", true},
> > +		{"image", true},
> > +		{"command", false},
> > +	}
> > +}
> > +
> > +func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
> > +	labelString, found := params["labels"]
> > +	var labels map[string]string
> > +	if found && len(labelString) > 0 {
> > +		labels = ParseLabels(labelString)
> > +	} else {
> > +		labels = map[string]string{
> > +			"name": params["name"],
>
> Selector interface won't work when I try to put it back into the api
> object (Selector is not map[string]string)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3424/files#r22897598>
> .
>